pysat/tests/test_meta.py

```diff
@@ -226,1375 +344,1474 @@ def test_init_labels_w_int_default(self):
+
+        assert len(meta_vals) == len(self.testInst.variables) - num_drop
+        for val in self.dval:
+            assert val not in meta_vals, \
+                "{:} not dropped from Meta".format(val.__repr__())
+        return
+
+    @pytest.mark.parametrize("num_keep", [0, 1, 3])
+    def test_meta_keep(self, num_keep):
+        """Test successful deletion of meta data for values not specified.
+
+        Parameters
+        ----------
+        num_keep : int
+            Number of variables to keep in a single go.
+
+        """
+
+        # Set meta data
+        self.set_meta(inst_kwargs={'platform': 'pysat', 'name': 'testing'})
+
+        # Get the data variables to drop
+        self.dval = self.testInst.variables[:num_keep]
+        testing.assert_list_contains(self.dval,
+                                     [val for val in self.meta.keys()])
 
-    @pytest.mark.parametrize("bad_key,bad_val,err_msg",
-                             [("col_names", [], "col_names must include"),
-                              ("filename", None, "Must provide an instrument"),
-                              ("filename", 5, "keyword name must be related"),
-                              ("filename", 'fake_inst',
-                               "keyword name must be related")])
-    def test_meta_csv_load_w_errors(self, bad_key, bad_val, err_msg):
-        """Test error handling when loading metadata from a csv file."""
+        # Drop the values
+        self.meta.keep(self.dval)
 
-        name = os.path.join(pysat.__path__[0], 'tests', 'cindi_ivm_meta.txt')
-        kwargs = {'filename': name, 'na_values': [],
-                  'keep_default_na': False, 'col_names': None}
-        kwargs[bad_key] = bad_val
-        with pytest.raises(ValueError) as excinfo:
-            pysat.Meta.from_csv(**kwargs)
-        assert str(excinfo.value).find('') >= 0
+        # Test the successful deletion of unspecified values
+        testing.assert_lists_equal(self.dval, [val for val in self.meta.keys()])
         return
 
-    # assign multiple values to default
-    def test_multiple_input_names_null_value(self):
-        """Test setting multiple input names to null."""
+    @pytest.mark.parametrize('use_method', [True, False])
+    def test_nan_metadata_filtered_netcdf4(self, use_method):
+        """Test that metadata set to NaN is excluded from netCDF output.
 
-        self.meta[['test1', 'test2']] = {}
-        assert self.meta['test1', 'units'] == ''
-        assert self.meta['test2', 'long_name'] == 'test2'
-        return
+        Parameters
+        ----------
+        use_method : bool
+            Use meta method and `export_nan` kwarg if True, use defaults
+            if False
 
-    def test_multiple_input_names_null_value_preexisting_values(self):
-        """Test setting multiple input names to null w/ pre-existing values."""
+        """
+        # TODO(#585): consider moving to class with netCDF tests
```

Comment:
I thought #585 was to add a print to netcdf test as part of cls_instrument_library.py.  This might need a new issue.